### ISSUE TITLE
Missing distributions

### ISSUE THREAD

**Issue by lostella** [awslabs/gluon-ts#467]
Some of the distributions which are currently missing are:
- [x] [Gamma](https://en.wikipedia.org/wiki/Gamma_distribution)
- [x] [Beta](https://en.wikipedia.org/wiki/Beta_distribution) (this can be implemented ad hoc, or can use Dirichlet internally, since it's a special case)
- [ ] [Inflated Beta](https://arxiv.org/pdf/0705.0700.pdf)

**Comment by jaheba** [comment]
Help wanted?

**Comment by lostella** [comment]
@jaheba indeed

**Comment by AaronSpieler** [comment]
Inflated Beta cannot be used because it has a non continuous PDF.

**Comment by AaronSpieler** [comment]
Everything addressed.

**Comment by AaronSpieler** [comment]
The PDF of the inflated Beta is separable, so the parameters can be optimised independently, which would allow us to implement it as a mixture of distributions.

**Comment by lostella** [comment]
Once #982 is merged, the inflated distributions should be a piece of 🍰

**Comment by PascalIversen** [comment]
Since currently, log_prob returns the pmf for discrete distributions and the pdf for continuous distributions, we can not mix discrete and continuous distributions if they have overlapping support.
We can not implement the `InflatedBeta`  and `InflatedGamma` as a naive `MixtureDistribution` between `Beta`/`Gamma` and the `Deterministic` because the distributions have overlapping support for some parameter ranges. (E.g. the beta distribution has infinite density at zero for alpha=0.5, beta=0.5)

**Comment by lostella** [comment]
@PascalIversen regarding Beta, the authors of https://www.researchgate.net/profile/Silvia_Ferrari/publication/1889063_Inflated_Beta_Distributions/links/09e41505c6e8562e88000000/Inflated-Beta-Distributions.pdf?origin=publication_detail don't seem to treat those cases differently (see eq 2 and fig 1). Maybe you can look into that, although they use a slightly different parametrization.

**Comment by PascalIversen** [comment]
Right, I see. Then the problem is not what I mentioned but rather a inconsistency/bug in the log_prob of the Beta. For alpha, beta < 1 it returns inf for 0 and 1 and not -inf.

**Comment by lostella** [comment]
In any case, eq (2) in the above paper suggests that the mixture density is in fact to be computed by cases, which is what @vafl suggested in #982
